test/js/bun/http/form-data-set-append.test.js

```diff
@@ -0,0 +1,65 @@
+import { expect, test } from "bun:test";
+
+// https://github.com/oven-sh/bun/issues/12325
+
+test("formdata set with File works as expected", async () => {
+  const expected = ["617580375", "text-notes1.txt"];
+
+  const server = Bun.serve({
+    port: 0,
+    fetch: async req => {
+      const data = await req.formData();
+      const chat_id = data.get("chat_id");
+      const document = data.get("document");
+      expect(chat_id).toEqual(expected[0]);
+      expect(document.name).toEqual(expected[1]);
+      return new Response("");
+    },
+  });
+
+  async function sendDocument(body) {
+    const response = await fetch(server.url, {
+      method: "POST",
+      body: body,
+    });
+    const text = await response.text();
+    return text;
+  }
+
+  const formDataSet = new FormData();
+  formDataSet.set("chat_id", expected[0]);
+  formDataSet.set("document", new File(["some text notes 1"], expected[1]));
+  await sendDocument(formDataSet);
+  server.stop();
+});
+
+test("formdata apppend with File works as expected", async () => {
+  const expected = ["617580376", "text-notes2.txt"];
+
+  const server = Bun.serve({
```

Comment:
nit: we can do `using` and that will automatically close the server without needing to manually call ``.stop`
```suggestion
  using server = Bun.serve({
```